CMakeLists.txt

```diff
@@ -0,0 +1,403 @@
+set(FPLLL_VERSION ${fplll_VERSION})
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  if(CMAKE_CXX_COMPILER_VERSION MATCHES "^14.")
+    message(STATUS "You're using ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}")
+  else()
+    message(WARNING "!! We recommend that you use clang-14 for this bootcamp. You're using ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}, a different version.")
+  endif()
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+  message(STATUS "You're using ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}")
+else()
+  message(WARNING "!! We recommend that you use clang-14 for this bootcamp. You're using ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}, which is not clang.")
+endif()
+
+# standard library use c++11
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_SYSROOT $ENV{SDKROOT})
+
+# Find system dependencies
+# GMP
+find_path(GMP_INCLUDE_DIR gmp.h PATHS /opt/homebrew/include /opt/homebrew/opt/gmp/include)
+find_library(GMP_LIBRARY NAMES gmp libgmp PATHS /opt/homebrew/lib /opt/homebrew/opt/gmp/lib)
+if (GMP_INCLUDE_DIR AND GMP_LIBRARY)
+  set(GMP_FOUND TRUE)
+  set(HAVE_LIBGMP 1)
+  set(LIBGMP_LIB "-lgmp")
+  message(STATUS "Found GMP: ${GMP_LIBRARY}")
+else ()
+  set(GMP_FOUND FALSE)
+  message(FATAL_ERROR "GMP not found")
+endif ()
+mark_as_advanced(GMP_INCLUDE_DIR GMP_LIBRARY)
+
+# MPFR
+find_path(MPFR_INCLUDE_DIR mpfr.h PATHS /opt/homebrew/include /opt/homebrew/opt/mpfr/include)
+find_library(MPFR_LIBRARY NAMES mpfr libmpfr PATHS /opt/homebrew/lib /opt/homebrew/opt/mpfr/lib)
+if (MPFR_INCLUDE_DIR AND MPFR_LIBRARY)
+  set(MPFR_FOUND TRUE)
+  set(HAVE_LIBMPFR 1)
+  set(LIBMPFR_LIB "-lmpfr")
+  message(STATUS "Found MPFR: ${MPFR_LIBRARY}")
+else ()
+  set(MPFR_FOUND FALSE)
+  message(FATAL_ERROR "MPFR not found")
+endif ()
+mark_as_advanced(MPFR_INCLUDE_DIR MPFR_LIBRARY)
+
+# QD (Optional)
+find_path(QD_INCLUDE_DIR qd/dd_real.h PATHS /opt/homebrew/include /opt/homebrew/opt/qd/include)
+find_library(QD_LIBRARY NAMES qd libqd PATHS /opt/homebrew/lib /opt/homebrew/opt/qd/lib)
+if (QD_INCLUDE_DIR AND QD_LIBRARY)
+  set(QD_FOUND TRUE)
+  set(FPLLL_WITH_QD 1)
+  set(LIBQD_LIB "-lqd")
+  message(STATUS "Found QD: ${QD_LIBRARY}")
+else ()
+  set(QD_FOUND FALSE)
+  message(WARNING "QD not found, proceeding without QD")
+endif ()
+mark_as_advanced(QD_INCLUDE_DIR QD_LIBRARY)
+
+find_package(Threads REQUIRED)
+include(CheckCXXSourceCompiles)
+if(Threads_FOUND)
+  set(HAVE_PTHREAD 1)
+  set(PTHREAD_LIB "-lpthread")
+  set(PTHREAD_CFLAGS "-pthread")
+  check_cxx_source_compiles("
+    #include <pthread.h>
+    int main() {
+        pthread_mutexattr_t attr;
+        return pthread_mutexattr_setprotocol(&attr, PTHREAD_PRIO_INHERIT);
+    }
+  " HAVE_PTHREAD_PRIO_INHERIT)
+else()
+  message(STATUS "pthread headers and libraries not found")
+endif()
+
+# Define project specific variables
+# TODO: (alex) revert
+# set(FPLLL_MAX_ENUM_DIM 256)
+set(FPLLL_MAX_ENUM_DIM 0)
```

Comment:
We should probably fix this. Is this commented out for configuration reasons?

Reply:
😆 ah! sry, i forgot to uncomment this, this was my local setup where I don't need this, thus turning them off for faster (re)compilation